modules/core/src/test/java/org/apache/ignite/internal/encryption/CacheGroupReencryptionTest.java

```diff
@@ -921,4 +946,52 @@ public IndexedObject(long id, String name) {
             return Objects.hash(name, id);
         }
     }
+
+    /**
+     * Finds first not reencrypted partition and returns tuple with cache group id, partition id and reencrypted page index.
+     */
+    private static GridTuple3<Integer, Integer, Integer> findFirstNotReencrypted(IgniteEx node, Iterable<String> cacheGrps) {
+        for (String cacheGrp : cacheGrps) {
+            int grpId = CU.cacheId(cacheGrp);
+
+            CacheGroupContext grp = node.context().cache().cacheGroup(grpId);
+
+            if (grp == null || !grp.affinityNode())
+                return null;
+
+            for (int p = 0; p < grp.affinity().partitions(); p++) {
+                long state = node.context().encryption().getEncryptionState(grpId, p);
```

Comment:
I think you can use `reencryptionPageIndex` here

Reply:
Not really, `reencryptionPageIndex` also checks the count and can return MAX_VALUE

Reply:
But you also check the count below, you can compare with MAX_VALUE instead